clippy_lints/src/floating_point_arithmetic.rs

```diff
@@ -297,60 +297,6 @@ fn check_powf(cx: &LateContext<'_>, expr: &Expr<'_>, args: &[Expr<'_>]) {
     }
 }
 
-fn check_powi(cx: &LateContext<'_>, expr: &Expr<'_>, args: &[Expr<'_>]) {
-    if let Some((value, _)) = constant(cx, cx.typeck_results(), &args[1]) {
-        if value == Int(2) {
-            if let Some(parent) = get_parent_expr(cx, expr) {
-                if let Some(grandparent) = get_parent_expr(cx, parent) {
-                    if let ExprKind::MethodCall(PathSegment { ident: method_name, .. }, _, args, _) = grandparent.kind {
-                        if method_name.as_str() == "sqrt" && detect_hypot(cx, args).is_some() {
-                            return;
-                        }
-                    }
-                }
-
-                if let ExprKind::Binary(
-                    Spanned {
-                        node: BinOpKind::Add, ..
-                    },
-                    lhs,
-                    rhs,
-                ) = parent.kind
-                {
-                    let other_addend = if lhs.hir_id == expr.hir_id { rhs } else { lhs };
-
-                    span_lint_and_sugg(
-                        cx,
-                        SUBOPTIMAL_FLOPS,
-                        parent.span,
-                        "square can be computed more efficiently",
-                        "consider using",
-                        format!(
-                            "{}.mul_add({}, {})",
-                            Sugg::hir(cx, &args[0], ".."),
-                            Sugg::hir(cx, &args[0], ".."),
-                            Sugg::hir(cx, other_addend, ".."),
-                        ),
-                        Applicability::MachineApplicable,
-                    );
```

Comment:
Why does this case also removed?

Reply:
It is just a special case for https://github.com/rust-lang/rust-clippy/blob/c8d52e35c10ca7cfa841309a7db4052344b710bf/clippy_lints/src/floating_point_arithmetic.rs#L403
Should we keep it?

Reply:
In the related issue, I found that the case of `x * x` was discussed, but I couldn't find the discussion of `mul_add`. So I don't know why this case needs to be removed as same.

If there is no reason, I think it would be better to keep it.